CRM/Core/Payment.php

```diff
@@ -508,7 +508,7 @@ public function getPreApprovalDetails($storedDetails) {
    */
   public function validatePaymentInstrument($values, &$errors) {
     CRM_Core_Form::validateMandatoryFields($this->getMandatoryFields(), $values, $errors);
-    if ($this->_paymentProcessor['payment_type'] == 1) {
+    if ($this->_paymentProcessor['payment_instrument_id'] == CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'payment_instrument_id', 'Credit Card')) {
```

Comment:
There are other types of Payments that the Dummy Processor processes, notably Pay Later. So I don't think this workaround is sufficient.

Reply:
If the contribution is pay later, `$this->_paymentProcessor['payment_instrument_id']` is 4 (check), so we don't try to validate the credit card.

Reply:
The issue is that `$this->_paymentProcessor['payment_type']` does not exist for pay later any more, so we need another way of checking if we are using a credit card.